src/parquet/properties.h

```diff
@@ -38,6 +39,101 @@ struct ParquetVersion {
 static int64_t DEFAULT_BUFFER_SIZE = 0;
 static bool DEFAULT_USE_BUFFERED_STREAM = false;
 
+class PARQUET_EXPORT ColumnEncryptionProperties {
+ public:
+  ColumnEncryptionProperties() = default;
+  ColumnEncryptionProperties(bool encrypt, std::string path)
+      : encrypt_(encrypt), path_(path), encrypted_with_footer_key_(encrypt) {}
+
+  bool encrypted() { return encrypt_; }
+  bool encrypted_with_footer_key() { return encrypted_with_footer_key_; }
+  std::string key() { return key_; }
+  std::string key_metadata() { return key_metadata_; }
+
+  void set_encryption_key(std::string key, uint32_t key_id = 0) {
+    std::string key_metadata =
+        key_id == 0 ? "" : std::string(reinterpret_cast<char*>(&key_id), 4);
+    set_encryption_key(key, key_metadata);
+  }
+
+  void set_encryption_key(std::string key, std::string key_metadata) {
+    if (!encrypt_) throw ParquetException("Setting key on unencrypted column: " + path_);
+    if (key.empty()) throw ParquetException("Null key for " + path_);
```

Comment:
Under what conditions would these situations occur?

Reply:
They occur when the constructor is called with encrypted = false or encrypted = true but key is empty. 
I reflected this function from java code.